src/solvers/nlopt_optimization_solver.C

```diff
@@ -78,16 +78,15 @@ double __libmesh_nlopt_objective(unsigned n,
     {
       if (solver->gradient_object != nullptr)
         {
-          solver->gradient_object->gradient(
-                                            *(sys.current_local_solution), *(sys.rhs), sys);
+          solver->gradient_object->gradient(*(sys.current_local_solution), *(sys.rhs), sys);
 
           // we've filled up sys.rhs with the gradient data, now copy it
           // to the nlopt data structure
           libmesh_assert(sys.rhs->size() == n);
 
           std::vector<double> grad;
           sys.rhs->localize_to_one(grad);
-          for (unsigned i=0; i<n; ++i)
+          for (auto i : IntRange<dof_id_type>(0, n))
```

Comment:
Good catch... but shouldn't we be changing the type of n to dof_id_type as well?

Reply:
Good catch, will fix this and the other one.

Reply:
So here we are somewhat constrained by the [nlopt interface](https://nlopt.readthedocs.io/en/latest/NLopt_Reference/#objective-function) itself which is expecting an `unsigned int`.
```
double f(unsigned n, const double* x, double* grad, void* f_data);
```
That probably means that we should `static_assert` that we aren't using the nlopt code when `sizeof(dof_id_type) != 4`...